docs/reference/architectures.rst

```diff
@@ -0,0 +1,183 @@
+Architectures
+=============
+
+snapcraft.yaml
+--------------
+
+The :doc:`architectures how-to</howto/architectures>` provides examples of how
+to use the ``architectures`` keyword.
+
+core22
+^^^^^^
+
+The root keyword ``architectures`` defines a list of ``build-on->build-for``
+pairs with the following syntax:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: [<arch 1>, <arch 2>]
+      build-for: [<arch 1>]
+    - build-on: [<arch 3>]
+      build-for: [<arch 4>]
```

Comment:
These are many-to-one mappings, right? e.g. I couldn't do:

```yaml
architectures:
  - build-on: amd64
    build-for: [amd64, riscv64]
```

Reply:
many-to-one for core22
many-to-many (multi-arch) for core20

I've made it more clear in the docs.